File: Zana-HMIS-API/api/api/src/main/java/com/orbix/api/domain/GeneralExamination.java
/**
 * 
 */
package com.orbix.api.domain;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Godfrey
 *
 */
@Entity
@Data 
@NoArgsConstructor 
@AllArgsConstructor
@Table(name = "general_examinations")
public class GeneralExamination {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String pressure;
	private String temperature;
	private String pulseRate;
	private String weight;
	private String height;
	private String bodyMassIndex;
	private String bodySurfaceArea;
	private String saturationOxygen;
	private String respiratoryRate;
	private String description;
	
	@OneToOne(targetEntity = Consultation.class, fetch = FetchType.LAZY,  optional = true)
    @JoinColumn(name = "consultation_id", nullable = true , updatable = true)
    @OnDelete(action = OnDeleteAction.NO_ACTION)	
    private Consultation consultation;
}

File: zana-hmis/src/app/domain/general-examination.ts
import { IDay } from "./day"
import { IUser } from "./user"
import { IConsultation } from "./consultation"

export interface IGeneralExamination {
    temperature: string
    id                  : any
    bodyMassIndex       : string
    bodyMassIndexComment  : string
    bodySurfaceArea     :string
    height              : string
    pressure            : string
    pulseRate           : string
    respiratoryRate     : string
    saturationOxygen    : string
    temparature         : string
    weight              : string
    description         : string

    consultation        : IConsultation

    
    createdBy           : IUser
    createdOn           : IDay
    createdAt           : Date

}